conftest.py

```diff
@@ -0,0 +1,19 @@
+"""Contains pytest hooks. The hooks defined here are used to modify the behavior of
+pytest during test collection and execution.
+"""
+
+import subprocess
+
+import pytest
+
+
+def pytest_collection_modifyitems(session, config, items) -> None:
+    """Run npm build only if e2e tests are being run. `items` is a list of test items to be
+    run.
+    """
+    # Check if running unit tests via -e2e parameter
+    if config.getoption("-k") == "e2e":
+        try:
+            subprocess.run(["npm", "run", "build"], check=True)
+        except subprocess.CalledProcessError as e:
+            pytest.exit(f"Failed to run npm build: {e}")
```

Comment:
This is failing to run on my Windows machine with a `FileNotFoundError`. I found out it's because the npm command should be `npm.cmd` in Windows.

I recommend using `npm.cmd` in Windows, and being explicit with what mode Webpack is to be run in.

(These changes require an `import os` at the top of the file)

```suggestion
    # Check if running unit tests via -e2e parameter
    if config.getoption("-k") == "e2e":
        try:
            npm_cmd = "npm.cmd" if os.name == "nt" else "npm"
            subprocess.run(
                [npm_cmd, "run", "build"],
                check=True,
                env={
                    **os.environ,
                    "WEBPACK_MODE": "development",
                },
            )
        except subprocess.CalledProcessError as e:
            pytest.exit(f"Failed to run npm build: {e}")
        except FileNotFoundError as e:
            pytest.exit(
                f"npm command not found (got: {e}). npm must be installed to run e2e tests."
            )

```

Reply:
I see, I didn't know that `npm.cmd` is actually the command that gets called for npm on Windows. It turns out `npm` does not work on Windows using `subprocess.run` because it does not use shell resolution by default. 

Would it be better to use `shutil.which("npm")` instead to look for the path of the `npm` executable on any given platform?

Addressed by:
https://github.com/NationalCentreTruthReconciliation/Secure-Record-Transfer/pull/344/commits/39739825fb63b7ba26674bcfb21814439e744482